tests/ops/functions/test_equal.py

```diff
@@ -107,6 +108,66 @@
 )
 
 
+PARAMETRIZED_MEASUREMENTS = [
+    qml.sample(qml.PauliY(0)),
+    qml.sample(wires=0),
+    qml.sample(),
+    qml.counts(qml.PauliZ(0)),
+    qml.counts(wires=[0, 1]),
+    qml.counts(wires=[1, 0]),
+    qml.counts(),
+    qml.density_matrix(wires=0),
+    qml.density_matrix(wires=1),
+    qml.var(qml.PauliY(1)),
+    qml.var(qml.PauliY(0)),
+    qml.expval(qml.PauliX(0)),
+    qml.expval(qml.PauliX(1)),
+    qml.probs(wires=1),
+    qml.probs(wires=0),
+    qml.probs(qml.PauliZ(0)),
+    qml.probs(qml.PauliZ(1)),
+    qml.state(),
+    qml.vn_entropy(wires=0),
+    qml.vn_entropy(wires=0, log_base=np.e),
+    qml.mutual_info(wires0=[0], wires1=[1]),
+    qml.mutual_info(wires0=[1], wires1=[0]),
+    qml.mutual_info(wires0=[1], wires1=[0], log_base=2),
+    qml.classical_shadow(wires=[0, 1]),
+    qml.classical_shadow(wires=[1, 0]),
+    qml.classical_shadow(wires=[0, 1], seed_recipes=False),
+    qml.shadow_expval(
+        H=qml.Hamiltonian(
+            [1.0, 1.0], [qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliX(0) @ qml.PauliX(1)]
+        ),
+        k=2,
+    ),
+    qml.shadow_expval(
+        H=qml.Hamiltonian(
+            [1.0, 1.0], [qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliX(0) @ qml.PauliX(1)]
+        )
+    ),
+    qml.shadow_expval(
+        H=qml.Hamiltonian(
+            [1.0, 1.0], [qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+        )
+    ),
+    qml.shadow_expval(
+        H=qml.Hamiltonian(
+            [1.0, 1.0], [qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+        ),
+        seed_recipes=False,
+    ),
+    MeasurementProcess(ObservableReturnTypes.Expectation, eigvals=[1, -1]),
+    MeasurementProcess(ObservableReturnTypes.Expectation, eigvals=[1, 2]),
+]
```

Comment:
The `PARAMETRIZED_MEASUREMENTS` list contains the functions that are currently defined in `measurements.py`.

When a new function is defined (i.e. a new measurement type), is there a way to ensure that a test would be added here for the new function? I am concerned because by looking at `measurements.py`, there is no way to tell `equal.py` exists. Perhaps a comment somewhere in `measurements.py` asking to add a test in `test_equal.py` can help?

Reply:
This is another reason of why we need to add an `__eq__` method to each class instead of using an external function. Doing this, the equal tests would be inside the `test_measurements.py` file and this problem would be solved.

Don't worry for now, given that we will probably start using the `__eq__` method.

Thank you for pointing this out!! 🙌